### ISSUE TITLE
Creating multiple subnets fails with "Token is currently in use for a different request"

### ISSUE THREAD

**Issue by tomzepf-oracle** [oracle/terraform-provider-oci#170]
## Description of the bug:

Regression with provider version v1.0.14, works fine with prior versions.

Creating 3 subnets at the same time fails with:

```
Error applying plan:

1 error(s) occurred:

* baremetal_core_subnet.public[2]: 1 error(s) occurred:

* baremetal_core_subnet.public.2: Status: 409; Code: Conflict; OPC Request ID: /6C0A5D4649AA45D4B6DE2227A6584609/543069731B5F4F04AB2E408FC9488D64; Message: Token is currently in use for a different request: ocid1.tenancy.oc1..aaaaaaaa6gtmn46bketftho3sqcgrlvdfsenqemqy3urkbthlpkos54a6wsa|ocid1.user.oc1..aaaaaaaasewdyptqgkuiutaccwfsspu6wwhaz35da5mdvyydtizx6zmpkfsq~ocid1.compartment.oc1..aaaaaaaad3lx3grr64g4s6i3s3moxf4pfgacz2rc42grki4up5rytmhoz7xa~Aey5duzSKQxVKQ3rFU1xpPUrQ8nhI5
```

## A self-contained terraform file that reproduces the issue:

(If this is not provided, your ticket will be closed)

[vcn.zip](https://github.com/oracle/terraform-provider-baremetal/files/1131146/vcn.zip)

**Comment by rcohenma** [comment]
I have not been able to reproduce this.

In v.1.0.14 we introduced automatic retry logic (to fix eventual consistency issues) and added a randomly generated retry token to each request (to ensure idempotency in our retries). The random generator is seeded with the current time. My guess is that you are running in an environment where time.Now().UnixNano() always return the same value and we are therefore always passing the same retry token for all of your requests.
https://docs.us-phoenix-1.oraclecloud.com/Content/API/Concepts/usingapi.htm#ten

If this is not the case could you set
export DEBUG=TRUE
export TF_LOG=DEBUG

and give us the logs when you get an error 

P.S. We are working on fixing the performance issue on destroys introduced in v1.0.14

**Comment by tomzepf-oracle** [comment]
I’ve verified the problem does NOT reproduce in Linux. But I can hit it every time on Windows 10 where I do most of my work. Maybe a known Go/Windows bug?

I’ve attached a log with DEBUG turned on as you suggested.

[terraform.apply.txt](https://github.com/oracle/terraform-provider-baremetal/files/1153282/terraform.apply.txt)

**Comment by tomzepf-oracle** [comment]
I’ve verified the problem does NOT reproduce in Linux. But I can hit it every time on Windows 10 where I do most of my work. Maybe a known Go/Windows bug?

 

I’ve attached a log with DEBUG turned on as you suggested.

 

Thanks,

 

     Tom

**Comment by tehmaspc** [comment]
Glad to see this post - since upgrading from .12 to .14 I have seen many errors for my TF module tests - namely around spinning up a VCN and then subsequent shape into the VCN. I get error messages like the OP above. I'm going to downgrade and take a closer look to see if that helps me and if I can provide more details shortly.

**Comment by rcohenma** [comment]
v1.0.16 should includes a fix for this. We cannot test on every possible environment so please let us know if you are still seeing this issue when the automatic retries are enabled.

**Comment by codycushing** [comment]
@tomzepf-oracle has the issue been resolved for you?

**Comment by tomzepf-oracle** [comment]
No, problem still exists, just harder to hit.

**Comment by rcohenma** [comment]
This should be fixed in release v2.0.5
If you continue to see this problem after upgrading, please reopen this issue.
